docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,433 @@
+that isolate a subset of the log outputs from the analyzer that should be
+processed. In this case it was important to only target this "deduped" log
+file, since otherwise the original `eve.json` would still have been processed
+and the error due to the presence of the duplicate fields would have halted the
+`brimcap load`.
+
+```
+analyzers:
+...
+  - cmd: suricata-wrapper.sh
+    globs: ["deduped*.json"]
+```
+
+What follows below the `globs:` setting is a Zed shaper configuration. Whereas
+the Zeek TSV logs contain Zed-compatible rich data types (timestamps, IP
+addresses, etc.), since Suricata's EVE logs are JSON, here we use this shaper
+config to assign better data types as the JSON is being converted for storage
+into the Zed Lake. Out-of-the-box, Brimcap automatically applies this same
+shaping configuration on the EVE output generated from its bundled Suricata.
+Here it's broken out and made part of the configuration YAML such that you can
+further modify it to suit your needs.
+```
+    shaper: |
+      type alert = {
+        timestamp: time,
+        event_type: bstring,
+        src_ip: ip,
+        src_port: port=(uint16),
+        dest_ip: ip,
+        dest_port: port=(uint16),
+        vlan: [uint16],
+        proto: bstring,
+        app_proto: bstring,
+        alert: {
+          severity: uint16,
+          signature: bstring,
+          category: bstring,
+          action: bstring,
+          signature_id: uint64,
+          gid: uint64,
+          rev: uint64,
+          metadata: {
+            signature_severity: [bstring],
+            former_category: [bstring],
+            attack_target: [bstring],
+            deployment: [bstring],
+            affected_product: [bstring],
+            created_at: [bstring],
+            performance_impact: [bstring],
+            updated_at: [bstring],
+            malware_family: [bstring],
+            tag: [bstring]
+          }
+        },
+        flow_id: uint64,
+        pcap_cnt: uint64,
+        tx_id: uint64,
+        icmp_code: uint64,
+        icmp_type: uint64,
+        tunnel: {
+          src_ip: ip,
+          src_port: port=(uint16),
+          dest_ip: ip,
+          dest_port: port=(uint16),
+          proto: bstring,
+          depth: uint64
+        },
+        community_id: bstring
+      }
+      filter event_type=alert | put . = shape(alert) | rename ts=timestamp
+```
+
+A full description of all that's possible with shaping configurations is beyond
+the scope of this article. However, this configuration is quite simple and can
+be described in brief.
+
+1. The `type alert` defines the names, [data types](https://github.com/brimdata/zed/blob/main/docs/formats/zson.md#33-primitive-values),
+and hierarchical locations of expected fields in the input records. Here we've
+defined a single "wide" shape for _all_ alerts we've known Suricata to
+generate, which is convenient because it allows Brim to easily display them in
+columnar format.
```

Comment:
I know it's been debated at times whether it's appropriate to use these wide schemas, particularly when it might results in records with excessive nulls because of alerts that don't use several fields. However, I've continued to come down on the side of erring on the side of "wide"-ness because we've had community users report it as a bug when data they're accustomed to seeing in columnar layouts in Brim suddenly lose their column headers because additional schemas have appeared. We did discuss once as a group some UX we could add in Brim to call a user's attention to the multiple schemas to make it easy for the user to see the data under each schema separately (e.g. perhaps under a separate "tab" per schema), and when that exists, I'd be more open to letting through more diverse schemas.